robot/impl/resource_manager.go

```diff
@@ -1065,7 +1065,9 @@ func (manager *resourceManager) createConfig() *config.Config {
 			conf.Remotes = append(conf.Remotes, *remoteConf)
 		} else if resName.API.IsComponent() {
 			conf.Components = append(conf.Components, resConf)
-		} else if resName.API.IsService() {
+		} else if resName.API.IsService() && !resConf.Equals(resource.Config{}) {
+			// Only append non-empty service configs. Internal services and remotes'
```

Comment:
I'm not sure this is correct. What if you specify a service with the attributes field set to nil?

Reply:
> What if you specify a service with the attributes field set to nil?

I _think_ this would only not append a service with an entirely zero-valued config (`Name` == "", `API == nil`, `Model` == nil, `Attributes` == nil, etc.). I do think you were right offline, and we can restrict this to only not append when the service is internal or contains a remote name. sorry for all the back and forth, let me know what you think of the new fix.